### PR TITLE
feat: add custom algolia search

### PR DESCRIPTION
**Pull request by neo773**
This PR migrates search to Algolia

You will need to apply [here](https://docsearch.algolia.com/) (it's free for open source projects) to get your credentials and replace in config.
Current credentials is demo 

https://github.com/tailcallhq/tailcallhq.github.io/assets/62795688/702624c4-a50d-46b2-b8d7-ed15d4ed6997

/claim #167

### CODE REVIEW

File: docusaurus.config.ts [hunk 1]
```diff
@@ -114,14 +114,13 @@ export default {
       respectPrefersColorScheme: false,
     },
     tableOfContents: {},
+    algolia: {
+      appId: "R2IYF7ETH7",
+      apiKey: "599cec31baffa4868cae4e79f180729b",
+      indexName: "docsearch",
+    },
```
Comment (tusharmath) [inline docusaurus.config.ts:121]: Need to use the following configuration.

```


<script type="text/javascript">


  docsearch({


    appId: "X27WDVHRQ3",


    apiKey: "35bc100f239853cd8a7195b23ed7393b",


    indexName: "tailcall",


    container: '### REPLACE ME WITH A CONTAINER (e.g. div) ###'


    debug: false 


  });

```

File: package.json [hunk 2]
```diff
@@ -20,6 +20,7 @@
     "@docusaurus/core": "3.1.1",
     "@docusaurus/plugin-google-gtag": "^3.1.1",
     "@docusaurus/preset-classic": "3.1.1",
+    "@docusaurus/theme-search-algolia": "^3.2.0",
```
Comment (tusharmath) [inline package.json:23]: We need to use the same UI that we use for lunr.

**Comment by tusharmath** [comment]
@neo773 Would you be continuing to work on this?

**Comment by neo773** [comment]
@tusharmath 
Yes,
It's 80% complete will be done by the end of this week.
Will create a new PR

https://github.com/tailcallhq/tailcallhq.github.io/compare/develop...neo773:tailcallhq.github.io:develop

**Pull request closed without merging**